gnosis/safe-react
# Show the nonce out of order warning

## What it solves
Resolves #2904

## How this PR fixes it
It iterates on the previous safeNonce out of order implementation and only shows the safe out of order warning if we deliberately edit the `safeNonce` and not in a normal create transaction flow despite of having pending transactions

## How to test it
1. Create a transaction that ends in the "Transaction Queue" list.
2. Create another transaction until the Review step. The review details of this transaction shall not show any warning.
3. Increment the `safeNonce` and you shall see the the safe nonce out of order warning.

## Screenshots
**On normal flow**
<img width="502" alt="Screen Shot 2021-11-18 at 19 46 31" src="https://user-images.githubusercontent.com/32431609/142477683-3898d3dc-3b82-4dc5-8bd8-ea0532767ce3.png">
**On incrementing the `safeNonce`**
<img width="503" alt="Screen Shot 2021-11-18 at 19 46 45" src="https://user-images.githubusercontent.com/32431609/142477701-8101c83b-1d00-4941-8932-396a6192e685.png">
